contracts/stellar-multicall/src/tests/test.rs

```diff
@@ -0,0 +1,248 @@
+#![cfg(test)]
+extern crate std;
+
+use soroban_sdk::testutils::Address as _;
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, IntoVal, Symbol, Val, Vec,
+};
+use stellar_axelar_std::events::{fmt_last_emitted_event, Event};
+use stellar_axelar_std::interfaces::OwnableInterface;
+use stellar_axelar_std::{assert_contract_err, interfaces, mock_auth, IntoEvent};
+
+use crate::error::ContractError;
+use crate::types::FunctionCall;
+use crate::{Multicall, MulticallClient};
+
+#[contract]
+pub struct TestTarget;
+
+#[derive(Debug, PartialEq, Eq, IntoEvent)]
+pub struct ExecutedEvent {
+    pub value: u32,
+}
+
+#[contractimpl]
+impl OwnableInterface for TestTarget {
+    fn owner(env: &Env) -> Address {
```

Comment:
why is this implemented manually? Use our derive macros. See how we do it for other contracts